pkg/es/config/config.go

```diff
@@ -238,21 +264,11 @@ func (c *Configuration) ApplyDefaults(source *Configuration) {
 	if c.AdaptiveSamplingLookback == 0 {
 		c.AdaptiveSamplingLookback = source.AdaptiveSamplingLookback
 	}
-	if c.NumShards == 0 {
-		c.NumShards = source.NumShards
-	}
-	if c.NumReplicas == 0 {
-		c.NumReplicas = source.NumReplicas
-	}
-	if c.PrioritySpanTemplate == 0 {
-		c.PrioritySpanTemplate = source.PrioritySpanTemplate
-	}
-	if c.PriorityServiceTemplate == 0 {
-		c.PriorityServiceTemplate = source.PriorityServiceTemplate
-	}
-	if c.PrioritySpanTemplate == 0 {
-		c.PriorityDependenciesTemplate = source.PriorityDependenciesTemplate
-	}
+
+	setFieldIfZero(&c.Indices.Spans, &source.Indices.Spans)
+	setFieldIfZero(&c.Indices.Services, &source.Indices.Services)
+	setFieldIfZero(&c.Indices.Dependencies, &source.Indices.Dependencies)
```

Comment:
how come this only covers template options, but not the date layout / rollover frequency fields?

Reply:
updated, I saw that the original logic only covered shards, so it was left out.